torch/csrc/jit/mobile/function.cpp

```diff
@@ -26,6 +27,57 @@ void tensorListConstruct(int num_inputs, Stack& stack) {
   drop(stack, num_inputs);
   push(stack, std::move(vals));
 }
+
+void tupleConstruct(int num_inputs, Stack& stack) {
+  std::vector<IValue> elems{
+    std::make_move_iterator(stack.end() - num_inputs),
+    std::make_move_iterator(stack.end())};
+  drop(stack, num_inputs);
+  push(stack, c10::ivalue::Tuple::create(std::move(elems)));
+}
+
+void tupleUnpack(int num_inputs, Stack& stack) {
+  auto tuple = pop(stack).toTuple();
+  if (tuple->elements().size() != num_inputs) {
+    AT_ERROR(
+       "Expected a tuple of ",
+       num_inputs,
+       " elements, but got ",
+       tuple->elements().size());
+  }
+  stack.insert(
+     stack.end(),
+     tuple->elements().begin(),
+     tuple->elements().end());
+}
+
+static const std::regex unsupported_options("\\{(.*?)\\}");
```

Comment:
nit: we can move this inside format() function if this is only used by it?

Reply:
Seems implementation is copied from torch/csrc/jit/register_special_ops.cpp?
What's our long term plan to support those ops in lite interpreter? Are we going to port all those ops to mobile?

Reply:
The difference between jit/register_special_ops.cpp (or register_prim_ops.cpp) and here is that in jit, there is reference to node. I think they can only be merged when jit is also using OPN + vararg ops. It may be feasible when the type specific is removed. @zdevito, correct me if I'm wrong.

Reply:
We should take care not to copy/paste operator implementations. In this care, for instance, we should have a single implementation of `format` and have the real and mobile interpreter call it.

Reply:
Similar for the above -- we should not start copy/pasting implementations of operators just because they are OPN instructions. Instead, we should put these `void op(int num_inputs, Stack& stack)` functions into a compilation unit and have both the mobile and regular interpreter call them.

Reply:
I'll make the essential functionalities shared between mobile and regular interpreter. I think that also address @ljk53 's concern.